generator/docs/NullOperator.md

```diff
@@ -0,0 +1,168 @@
+
+```
+{
+    "if": {
+        {
+            "field": "field1",
+            "is": "equalTo",
+            "value": 5
+        }
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    }
+}
+```
+
+The generator will complete the `if` constraint as follows, to ensure the constraint is fully balanced:
+
+```
+{
+    "if": {
+        {
+            "field": "field1",
+            "is": "equalTo",
+            "value": 5
+        }
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    },
+    "else": {
+        {
+            "not": {
+                "field": "field1",
+                "is": "equalTo",
+                "value": 5
+            }
+        }
+    }
+}
+```
+
+This expression does not prevent the consequence (the `then` constraints) from being considered when `field1` has no value. Equally it does not say anything about the alternative consequence (the `else` constraints). As such both outcomes can be considered at any time.
+
+The solution to this is to express the `if` constraint as follows. This is not 'auto completed' for profiles as it would remove functionality that may be intended, it must be explicitly included in the profile.
+
+```
+{
+    "if": {
+        "allOf": [
+            {
+                "field": "field1",
+                "is": "equalTo",
+                "value": 5
+            },
+            {
+                "not": {
+                    "field": "field1",
+                    "is": "null"
+                }
+            }
+        ]
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    }
+}
+```
+
+The generator will complete the `if` constraint as follows, to ensure the constraint is fully balanced:
+
+```
+{
+    "if": {
+        "allOf": [
+            {
+                "field": "field1",
+                "is": "equalTo",
+                "value": 5
+            },
+            {
+                "not": {
+                    "field": "field1",
+                    "is": "null"
+                }
+            }
+        ]
+    },
+    "then": {
+        {
+            "field": "field2",
+            "is": "equalTo",
+            "value": "a"
+        }
+    },
+    "else": {
+        "anyOf": [
+            {
+                "not": {
+                    "field": "field1",
+                    "is": "equalTo",
+                    "value": 5
+                }
+            },
+            {
+                "field": "field1",
+                "is": "null"
+            }
+        ]
+    }
+}
+```
+
+If this case the `then` constraints will only be value when `field1` has a value, otherwise it will fire in either of the `else` cases, but nevertheless is confined to this scope.
+
+### Examples:
+
+* A field that is non-nullable (in SQL terminology)<br />
```

Comment:
Instead of describing our functionality in terms of SQL terminology, you could present this information as a use case study - "You're trying to populate a SQL table, here's what you'd do for each of your fields".

Reply:
Section updated